Source/WebCore/Modules/notifications/Notification.cpp

```diff
@@ -57,6 +58,23 @@ namespace WebCore {
 
 WTF_MAKE_ISO_ALLOCATED_IMPL(Notification);
 
+static Lock nonPersistentNotificationMapLock;
+static HashMap<UUID, Notification*>& nonPersistentNotificationMap() WTF_REQUIRES_LOCK(nonPersistentNotificationMapLock)
+{
+    static NeverDestroyed<HashMap<UUID, Notification*>> map;
```

Comment:
If WeakPtrFactory were thread-safe (i.e., if we had a version of WeakPtr that coordinated with ThreadSafeRefCounted), then we could use WeakHashMap here. More motivation to do that, I guess.